File: python-operator/controller/deployments.py
import yaml

def make_deployment(name, label, image):
    return yaml.safe_load(f"""
apiVersion: apps/v1
kind: Deployment
metadata:
  name: {name}
spec:
  replicas: 1
  selector:
    matchLabels:
      app: {label}
  template:
    metadata:
      labels:
        app: {label}
    spec:
      imagePullSecrets:
      - name: gitlab-registry
      containers:
      - name: {label}
        image: "{image}"
""")

def get_deploymail(naam):
    return make_deployment(f"mail-deployment-{naam}", f"mail-deployment-{naam}", "registry.gitlab.warpnet.nl/securedropzone/securedropzone-e2ee/mail:1.7.0")

def get_deployweb(naam):
    return make_deployment(f"web-deployment-{naam}", f"web-deployment-{naam}", "registry.gitlab.warpnet.nl/securedropzone/securedropzone-e2ee/web:1.7.0")

def get_deploysms(naam):
    return make_deployment(f"sms-deployment-{naam}", f"sms-deployment-{naam}", "registry.gitlab.warpnet.nl/securedropzone/securedropzone-e2ee/sms:1.7.0")

def get_deploystorage(naam):
    return make_deployment(f"storage-deployment-{naam}", f"storage-deployment-{naam}", "registry.gitlab.warpnet.nl/securedropzone/securedropzone-e2ee/storage:1.7.0")

File: sdz-operator/internal/controller/deployments.go
package controller

import (
	securedropzonev1 "gitlab.warpnet.nl/securedropzone/sdz-operator/api/v1"
	appsv1 "k8s.io/api/apps/v1"
	corev1 "k8s.io/api/core/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

// Helper functie om een Deployment resource op te bouwen
func newDeployment(name string, replicas int32, label string, image string, sdz *securedropzonev1.SecureDropzone) appsv1.Deployment {
	return appsv1.Deployment{
		ObjectMeta: metav1.ObjectMeta{
			Name:      name,
			Namespace: sdz.Namespace,
		},
		Spec: appsv1.DeploymentSpec{
			Replicas: &replicas,
			Selector: &metav1.LabelSelector{
				MatchLabels: map[string]string{
					"app": label,
				},
			},
			Template: corev1.PodTemplateSpec{
				ObjectMeta: metav1.ObjectMeta{
					Labels: map[string]string{
						"app": label,
					},
				},
				Spec: corev1.PodSpec{
					Containers: []corev1.Container{
						{
							Name:  label,
							Image: image,
						},
					},
					ImagePullSecrets: []corev1.LocalObjectReference{
						{
							Name: "gitlab-registry",
						},
					},
				},
			},
		},
	}
}
